ldap_auth_provider.py

```diff
@@ -57,11 +58,15 @@ class _LdapConfig(object):
     default_domain: Optional[str] = None
 
 
-class LdapAuthProvider(object):
+SUPPORTED_LOGIN_TYPE: str = "m.login.password"
```

Comment:
Is this type hint necessary? Isn't it implied?

Reply:
Subjective style maybe, but I'm a fan of putting type annotations on top-level fields